### ISSUE TITLE
Asymmetry of, and working with Value, None, and Blank

### ISSUE THREAD

**Issue by attack68** [marshmallow-code/webargs#323]
Please do correct me if I'm wrong but there is often an asymmetry between webform input fields at initialisation and a webform input fields prepopulated from a Database (or touched by user), which causes use-cases with validating the concepts of `Value`, `None` and `""` (Blank).

For example a Webform might contain a text input (validated for a numeric type), which is initialised with a `null` value in JavaScript. On submit (without touching this input field) `webargs` can parse this with the `allow_none` flag set to `True`. Entered into a database a `None` value will evoke the `NULL` entry with specific meaning, or in other cases default values can be configured at time of database entry.

Now suppose that the user has touched the field or that it was prepopulated with a `Value` and the user removes it creating, instead,  a `""` entry.

While `webargs` will parse this, the fields [`fields.Float()`, `fields.Int()`] throw errors: "not a valid [number, integer]", even when `allow_blank=True`.

Thus it leaves the requirement of having to either implement JavaScript on the client side to ensure that `""` is converted to `null` before submitting the form for `webargs` to parse, or what might be a better pattern in some  to allow a specific option to cast `""` to `None` so that this is consistent. Consider the option:

`cast_blank_to_none=False`: no change to existing behaviour
`cast_blank_to_none=True`: if data='': data=None, before any further parsing.

**TLDR;**
Can `webargs` cast `""` (blank) to `None` so that a blank Numeric/Int input form is allowed, without the need to validate or augment userinput on the Javascript side.

**Comment by sloria** [comment]
We won't be adding this into webargs core, as this is more an issue with the deserialization behavior, which is in marshmallow's purview. There's a discussion about this here: https://github.com/marshmallow-code/marshmallow/issues/713
